elastic/kibana
# [Alert details page][Custom threshold] Add warning threshold info

Fixes #183628

## Summary

This PR adds warning threshold to the Threshold component in the alert details page. It also fixes the issue with range threshold.

|Warning info|Storybook|
|---|---|
|![image](https://github.com/elastic/kibana/assets/12370520/c48a6fa8-4df6-4f1e-bf7b-9f9d0b2b85be)|![image](https://github.com/elastic/kibana/assets/12370520/8084e27e-424a-48db-9eef-256190c96b89)|

### 🧪 How to test
- Enable `xpack.observability.unsafe.alertDetails.metrics.enabled: true` feature flag.
- Generate data with the following command
```
node x-pack/scripts/data_forge.js \
  --events-per-cycle 7 \
  --lookback now-20m \
  --install-kibana-assets \
  --dataset fake_hosts
```
- Adjust the Inventory setting and add `kbn-data-forge*` to the Metric indices
- Create a metric threshold rule with both alert and warning thresholds set
- Go to the alert details page and verify the warning info as stated above

#### Storybook
You can also check the [Threshold component storybook](https://ci-artifacts.kibana.dev/storybooks/pr-184571/787578147f949823baa7bc3b421be5ba0f948088/infra/index.html?path=/story/infra-alerting-threshold--default).
